pkg/fqdn/cache_test.go

```diff
@@ -1109,3 +1110,171 @@ func (ds *DNSCacheTestSuite) TestOverlimitPreferNewerEntries(c *C) {
+			if len(zombies[i].Names) > len(zombies[j].Names) {
+				logFailure(t, zombies, "len(names)", i, j)
+				t.Fatalf("order wrong: len(names): %v is longer than %v", zombies[i].Names, zombies[j].Names)
+			}
+		}
+	}
+}
+
+func Test_sortZombieMappingSlice(t *testing.T) {
+	// Create three moments in time, so we can have before, equal and after.
+	moments := []time.Time{
+		time.Date(2001, time.January, 1, 1, 1, 1, 0, time.Local),
+		time.Date(2002, time.February, 2, 2, 2, 2, 0, time.Local),
+		time.Date(2003, time.March, 3, 3, 3, 3, 0, time.Local),
+	}
+
+	// Couple of edge cases/hand-picked scenarios. To be complemented by the
+	// randomly generated ones, below.
+	type args struct {
+		zombies []*DNSZombieMapping
+	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			"empty",
+			args{zombies: nil},
+		},
+		{
+			"single",
+			args{zombies: []*DNSZombieMapping{{
+				Names:           []string{"test.com"},
+				AliveAt:         moments[0],
+				DeletePendingAt: moments[1],
+			}}},
+		},
+		{
+			"swapped alive at",
+			args{zombies: []*DNSZombieMapping{
+				{
+					AliveAt: moments[2],
+				},
+				{
+					AliveAt: moments[0],
+				},
+			}},
+		},
+		{
+			"equal alive, swapped delete pending at",
+			args{zombies: []*DNSZombieMapping{
+				{
+					AliveAt:         moments[0],
+					DeletePendingAt: moments[2],
+				},
+				{
+					AliveAt:         moments[0],
+					DeletePendingAt: moments[1],
+				},
+			}},
+		},
+		{
+			"swapped equal times, tiebreaker",
+			args{zombies: []*DNSZombieMapping{
+				{
+					Names:           []string{"test.com", "test2.com"},
+					AliveAt:         moments[0],
+					DeletePendingAt: moments[1],
+				},
+				{
+					Names:           []string{"test.com"},
+					AliveAt:         moments[0],
+					DeletePendingAt: moments[1],
+				},
+			}},
+		},
+	}
+
+	// Generate zombie mappings which cover all cases of the two times
+	// being either moment 0, 1 or 2, as well as with 0, 1 or 2 names.
+	names := []string{"example.org", "test.com"}
+	nMoments := len(moments)
+	allMappings := make([]*DNSZombieMapping, 0, nMoments*nMoments*nMoments)
+	for _, mi := range moments {
+		for _, mj := range moments {
+			for k := range moments {
+				m := DNSZombieMapping{
+					AliveAt:         mi,
+					DeletePendingAt: mj,
+					Names:           names[:k],
+				}
+				allMappings = append(allMappings, &m)
+			}
+		}
+	}
+
+	// Five random tests:
+	for i := 0; i < 5; i++ {
+		ts := make([]*DNSZombieMapping, len(allMappings))
+		copy(ts, allMappings)
+		rand.Shuffle(len(ts), func(i, j int) {
+			ts[i], ts[j] = ts[j], ts[i]
+		})
+		tests = append(tests, struct {
+			name string
+			args args
+		}{
+			name: "Randomised sorting test",
+			args: args{
+				zombies: ts,
+			},
+		})
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ol := len(tt.args.zombies)
+			sortZombieMappingSlice(tt.args.zombies)
+			if len(tt.args.zombies) != ol {
+				t.Fatalf("length of slice changed by sorting")
+			}
```

Comment:
Nit. To be super picky, one could also check that the elements match, in addition to the slice length (e.g., through `assert.ElementsMatch`). Feel free to ignore this comment though, as with the current implementation it would test that the go sort method works.

Reply:
True, but yeah as you said we'd just be testing Go's sort - I'll leave it.